internal/impl/kafka/enterprise/topic_crud.go

```diff
@@ -56,7 +57,40 @@ func createTopic(ctx context.Context, topic string, replicationFactorOverride bo
 		}
 	}
 
-	if _, err := outputAdminClient.CreateTopic(ctx, partitions, rp, nil, topic); err != nil {
+	topicConfigs, err := inputAdminClient.DescribeTopicConfigs(ctx, topic)
+	if err != nil {
+		return fmt.Errorf("failed to fetch configs for topic %q from source broker: %s", topic, err)
+	}
+
+	rc, err := topicConfigs.On(topic, nil)
+	if err != nil {
+		return fmt.Errorf("failed to fetch configs for topic %q from source broker: %s", topic, err)
+	}
+
+	destinationConfigs := make(map[string]*string)
+	for _, c := range rc.Configs {
+		// Source: https://docs.redpanda.com/current/reference/properties/topic-properties/
+		if c.Key == "cleanup.policy" ||
+			c.Key == "flush.bytes" ||
+			c.Key == "flush.ms" ||
+			c.Key == "initial.retention.local.target.ms" ||
+			c.Key == "retention.bytes" ||
+			c.Key == "retention.ms" ||
+			c.Key == "segment.ms" ||
+			c.Key == "segment.bytes" ||
+			c.Key == "compression.type" ||
+			c.Key == "message.timestamp.type" ||
+			c.Key == "max.message.bytes" ||
+			c.Key == "replication.factor" ||
+			c.Key == "write.caching" ||
+			c.Key == "redpanda.iceberg.mode" ||
+			strings.HasPrefix(c.Key, "redpanda.remote.") {
```

Comment:
I'm not sure we want to mirror `redpanda.remote.*` but just want to take defaults in the destination cluster. The thing is in Redpanda Cloud we never want this to be off, and the default is always on. Maybe we should just not mirror it...

Reply:
Sounds good, I removed `redpanda.remote.` from that list.